src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -150,58 +146,66 @@ class ParseLiveQueryServer {
           // Check CLP
           const op = this._getCLPOperation(subscription.query);
           let res = {};
-          this._matchesCLP(classLevelPermissions, message.currentParseObject, client, requestId, op)
-            .then(() => {
-              // Check ACL
-              return this._matchesACL(acl, client, requestId);
-            })
-            .then(isMatched => {
-              if (!isMatched) {
-                return null;
-              }
-              res = {
-                event: 'delete',
-                sessionToken: client.sessionToken,
-                object: deletedParseObject,
-                clients: this.clients.size,
-                subscriptions: this.subscriptions.size,
-                useMasterKey: client.hasMasterKey,
-                installationId: client.installationId,
-                sendEvent: true,
-              };
-              return maybeRunAfterEventTrigger('afterEvent', className, res);
-            })
-            .then(() => {
-              if (!res.sendEvent) {
-                return;
-              }
-              if (res.object && typeof res.object.toJSON === 'function') {
-                deletedParseObject = res.object.toJSON();
-                deletedParseObject.className = className;
+          try {
+            await this._matchesCLP(
```

Comment:
I'm not sure if it is a good idea. `requestIds` will be processed in series and it can bring performance problems. If you want to use async/await, maybe you should get rid of the `for`.

Reply:
Ahh yep. So use `Promise.all` instead?

Reply:
I am not sure if you need even to `await Promise.all` since the current version does not return the promise. `requestsIds.forEach(async requestId => { ... })` should be enough.